src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +18,36 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+        } catch (HibernateException e) {
+            throw new DataProcessingException("Cannot open session.", e);
```

Comment:
The exception message 'Cannot open session.' might be misleading since the exception is caught after the session is opened. Consider changing it to something more general like 'Error during transaction' to cover all possible Hibernate exceptions.